lib/dat-worker-pool/worker.rb

```diff
@@ -50,9 +50,16 @@ def initialize(runner, queue)
       end
 
       def work(work_item)
-        dwp_run_callback('before_work', work_item)
-        work!(work_item)
-        dwp_run_callback('after_work', work_item)
+        @dwp_runner.mark_worker_unavailable(self)
+        dwp_run_callback 'on_unavailable'
```

Comment:
@jcredding I think you need to move this run callback line into the begin..rescue so that if an error occurs the work gets marked available, right?